### ISSUE TITLE
Considering `.arb` a ruby file leads to 'frozen_string_literal' warnings

### ISSUE THREAD

**Issue by tagliala** [rubocop/rubocop#13684]
As per https://github.com/rubocop/rubocop/pull/5591, `.arb` files are considered Ruby extensions

However, they work as templates like `.erb` or `.haml`

In fact, they are being using in this way: https://github.com/activeadmin/arbre/blob/97da46273c0897aa5d695be56a055c74fb7e6085/lib/arbre/rails/template_handler.rb

--------

## Expected behavior

- Do not consider `.arb` Ruby files; or
- Disable `Style/FrozenStringLiteralComment` by default for that extension

## Actual behavior
```
$ cat index.html.arb 
h1 'Hello World'

$ rubocop --debug index.html.arb 
For ./: Default configuration from ~/.rvm/gems/ruby-3.4.1/gems/rubocop-1.70.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning ,/index.html.arb
Loading cache from ~/.cache/rubocop_cache/path/to/cache
C

Offenses:

index.html.arb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
h1 'Hello World'
^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.06852099997922778 seconds
```

## Steps to reproduce the problem

1. `echo "h1 'Hello World'" > index.html.arb`
2. `rubocop index.html.arb`
3. Fix the issue
4. Create a file `arbre_frozen.rb` with the following content:
  ```rb
  # frozen_string_literal: true
  
  require 'bundler/inline'
  
  begin
    gemfile(true) do
      source 'https://rubygems.org'
  
      gem 'arbre'
      gem 'rails'
    end
  
    require 'arbre'
    require 'arbre/railtie'
  rescue Gem::LoadError => e
    puts "\nMissing Dependency:\n#{e.backtrace.first} #{e.message}"
  rescue LoadError => e
    puts "\nError:\n#{e.backtrace.first} #{e.message}"
    exit 1
  end
  
  class FileTemplate
    attr_reader :source
  
    def initialize(file_path)
      @source = File.read(file_path)
    end
  end
  
  def render_arb_file(file_path, assigns = {})
    template = FileTemplate.new(file_path)
    handler = Arbre::Rails::TemplateHandler.new
    code = handler.call(template)
    eval(code, binding)
  end
  
  render_arb_file 'index.html.arb'
  ```
5. Run `$ RUBYOPT='-w' ruby arbre_frozen.rb`

Check the output:

```
(eval at arbre_test.rb:36):2: warning: 'frozen_string_literal' is ignored after any tokens
```

## RuboCop version

```
1.70.0 (using Parser 3.3.6.0, rubocop-ast 1.37.0, analyzing as Ruby 2.7, running on ruby 3.4.1) [x86_64-darwin24]
```

Ref: activeadmin/activeadmin#8600

**Comment by dvandersluis** [comment]
@tagliala Is `Style/FrozenStringLiteralComment` the only cop that should be excluded for `.arb` files? (I'm not familiar with them or activeadmin). If so we could just exclude `**/*.arb` from the default config for that cop.

**Comment by tagliala** [comment]
Hi,

> Is Style/FrozenStringLiteralComment the only cop that should be excluded for .arb files? 

Not sure, we do not have reports about cops that are interfering with `.arb`. We've noticed that one in the effort of reducing the number of warnings emitted by activeadmin

> If so we could just exclude **/*.arb from the default config for that cop.

Yes, if that is possible and it is used by other cops it would be appreciated.

It would be also good to have some feedback from David Rodríguez, which is also the author of #5591 (I'm not going to mention again, because I did that in the activeadmin repo, maybe it is not a good time)

**Comment by koic** [comment]
Active Admin's arb files appear to be Ruby itself, similar to jb, rather than a non-Ruby template language like ERB or Haml.
https://github.com/amatsuda/jb

Therefore, there doesn't seem to be a reasonable justification to disable the `Style/FrozenStringLiteralComment` cop by default for the arb extension. If disabling is necessary, it should be handled on the user's side. For this reason, I'm closing this issue. Thank you.

**Comment by tagliala** [comment]
@koic thanks for the answer

>  there doesn't seem to be a reasonable justification 

You mean that the suggestion to add a `# frozen_string_literal: true` comment on top of a .arb which

1. Produces a warning
2. Does not have effect

Is not a good reason to disable that specifc cop by default?

**Comment by Earlopain** [comment]
It's not possible to use magic comments in such cases since the comment _must_ appear before code and that seems to never be the case for these, somewhere ActiveAdmin adds a prelude I guess. Here's a related rails PR for `erb` https://github.com/rails/rails/pull/43725 where they properly handle adding the comment at the framework level.

I agree that this should be handled by RuboCop if it natively supports `.arb`.

**Comment by tagliala** [comment]
Thanks everyone for the precious insights
